src/modules/dome.wren

```diff
@@ -45,6 +45,24 @@ class Version {
 
 class Process {
   foreign static f_exit(n)
+  foreign static f_args
+
+  static arguments {Process.f_args}
+  static arguments(needle) {
+    var args = Process.arguments
+    var result = null
+    Fiber.new {
```

Comment:
I don't think we need a whole fiber just to abort a loop early. You can do:

```
for (value in args) {
  if (value.contains(needle)) {
    result = value
  }
}
```
However, I have concerns for two reasons: 
1) any argument could "contain" the needle and it would be caught here. You probably want `value.startsWith("--%(needle)")` for the specific effect you are trying to achieve.

2) this doesn't support "short" options: `--edit` could be just `-e` depending on what you wanted to enable. This is something optparse allows.

Command line processing has a lot of edge cases.

Reply:
I think for the sake of minimalism. Is fine for DOME just returning the whole array of arguments. And leave the parsing to the final user 👍

Reply:
The `--` is something that could be ditched since it conflicts with optparse

the convention could be that everything after the filename is part of the args 💯 

`./dome main.wren edit=tracks/1.dat new`

Reply:
The optparse library is what handles the `--`, so we can't easily ditch it. I was asking in the discord whether most people actually specify a particular script name, because we could easily add a named option for that, and it would fix the behaviour to be consistent with what optparse expects.

Reply:
Well by ditching `--` I mean only for the custom arguments.

Example: `./dome -d --record=data.gif main.wren edit=tracks/1.dat new`
This would be read as `[./dome, -d, --record=data.gif, main.wren, edit=tracks/1.dat, new]`

So you can say say ever argument that starts with `-` or `--` is a DOME argument. And everything without (after the filename) it can be considered a custom argument.

Reply:
Sure, although we can easily say: "anything without `-`, `--` and anything following a lone `--` are arguments available for usercode to use". 

But for any of this to work reliably, we need to change how DOME selects it's main file, so that it's a named argument. It would need to be something like `./dome -f main.wren ...` or `./dome --file=main.wren ...`, although the name/shortening are up for debate. The alternative is to swap out `optparse` for a different mechanism of parsing command line arguments entirely.